### ISSUE TITLE
Windows Support

### ISSUE THREAD

**Issue by mloskot** [sibprogrammer/xq#30]
**Problem Statement**

Would it be possible to release Windows binary too?

If there is one available, then I'm  willing to submit package `sibprogrammer.xq` to https://github.com/microsoft/winget-pkgs 

I think it would be great if `xq` is available via `winget install --id sibprogrammer.xq` on Windows and becomes part of the tools company together with https://github.com/stedolan/jq and https://github.com/mikefarah/yq which are already packaged for winget.

**Comment by sibprogrammer** [comment]
I agree it sounds reasonable to support the Windows platform as well.
I can't give any promises because have no permanent access to Win-based machines (but it's not a big deal to find, I hope), and `xq` uses some terminal-specific tricks (I also believe it is solvable). So sooner or later, such support will be added.

As for `winget` I didn't hear about it before. Meanwhile, Win-focused coworkers mention `chocolatey` rather often.  What is more popular? Or maybe one ecosystem integrates the parts of another (like it happens with `brew` and `macports` on Mac)?

**Comment by mloskot** [comment]
> What is more popular?

I don't have any stats to share. Chocolatey is definitely very popular, it's been around for long time.
But, winget is gaining popularity as it is an official package manager for Windows and by Microsoft, e.g.:
- https://learn.microsoft.com/en-gb/windows/package-manager/winget/
- https://learn.microsoft.com/en-gb/visualstudio/install/use-command-line-parameters-to-install-visual-studio

**Comment by sibprogrammer** [comment]
Ok, I got it. Winget from this point of view looks very attractive. @mloskot thank you for the clarification!

**Comment by orgcontrib** [comment]
> I agree it sounds reasonable to support the Windows platform as well. I can't give any promises because have no permanent access to Win-based machines (but it's not a big deal to find, I hope), and `xq` uses some terminal-specific tricks (I also believe it is solvable). So sooner or later, such support will be added.
> 
> As for `winget` I didn't hear about it before. Meanwhile, Win-focused coworkers mention `chocolatey` rather often. What is more popular? Or maybe one ecosystem integrates the parts of another (like it happens with `brew` and `macports` on Mac)?

`winget` is the official tool (should always be the 1st choice).
`scoop` is the next best choice (in my humble opinion).
[scoop.sh](https://scoop.sh)

**Comment by Utopiah** [comment]
FWIW based on the target audience and the amount of work required for maintainers who might not target Windows machine I'd start by insuring there is a binary that works with Windows Subsystem for Linux (WSL) and would thus also benefit from the existing `curl -sSL https://bit.ly/install-xq | sudo bash` installation solution.

**Comment by sergeevabc** [comment]
`xq` does not work under Windows 7 since you switched to @golang 1.21

``` console
$ xq.exe
Exception 0xc0000005 0x8 0x0 0x0
PC=0x0

runtime.asmstdcall()
        /opt/hostedtoolcache/go/1.21.5/x64/src/runtime/sys_windows_amd64.s:65 +0x75 fp=0x22fca0 sp=0x22fc80 pc=0x46bf15
rax     0x0
rbx     0xbae720
rcx     0xc01b40
rdi     0x7fffffde000
rsi     0x22fea0
rbp     0x22fde0
rsp     0x22fc78
r8      0x0
r9      0x22fee0
r10     0xbd3a78
r11     0x21
r12     0x22fec0
r13     0x1
r14     0xbae0c0
r15     0x0
rip     0x0
rflags  0x10293
cs      0x33
fs      0x53
gs      0x2b
```

**Comment by sibprogrammer** [comment]
@sergeevabc Thank you for notifying about this! Unfortunately, they've decided to stop supporting Windows 7. 
But probably I can organize a separate pipeline for building Windows binaries using the previous version of Go.
